# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1231984879 18000
#      Wed Jan 14 21:01:19 2009 -0500
# Node ID c2c018d7c5018b86029e104a303f0600118aead4
# Parent  81d6ab3ac93ccf719684db4b22c93a9f5626c48f
__go_draw_axes__.m: __go_draw_axes__.m: Support non-cell ticklabels.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-01-14  Ben Abbott  <bpabbott@mac.com>
+
+	* plot/__go_draw_axes__.m (ticklabel_to_cell): New function.
+	Use it to handle non-cell ticklabels.
+
 2009-01-14  Søren Hauberg  <hauberg@gmail.com>
 
 	* general/diff.m, general/logspace.m, general/nextpow2.m,
 	linear-algebra/commutation_matrix.m,
 	linear-algebra/duplication_matrix.m, linear-algebra/expm.m,
 	miscellaneous/bincoeff.m, miscellaneous/list_primes.m,
 	optimization/fsolve.m, plot/subplot.m, polynomial/pchip.m,
 	polynomial/polyout.m, polynomial/residue.m, polynomial/spline.m,
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -1373,16 +1373,20 @@ function __gnuplot_write_data__ (plot_st
       endif
     endif
   endif
 
 endfunction
 
 function do_tics (obj, plot_stream, ymirror, mono)
 
+  obj.xticklabel = ticklabel_to_cell (obj.xticklabel);
+  obj.yticklabel = ticklabel_to_cell (obj.yticklabel);
+  obj.zticklabel = ticklabel_to_cell (obj.zticklabel);
+
   [fontname, fontsize] = get_fontname_and_size (obj);
 
   if (strcmpi (obj.xaxislocation, "top"))
     do_tics_1 (obj.xtickmode, obj.xtick, obj.xminortick, obj.xticklabelmode,
 	       obj.xticklabel, obj.xcolor, "x2", plot_stream, true, mono,
 	       "border", obj.tickdir, fontname, fontsize, obj.interpreter);
     do_tics_1 ("manual", [], "off", obj.xticklabelmode, obj.xticklabel,
 	       obj.xcolor, "x", plot_stream, true, mono, "border",
@@ -1523,16 +1527,26 @@ function do_tics_1 (ticmode, tics, mtics
     if (strcmp (mtics, "on"))
       fprintf (plot_stream, "set m%stics 5;\n", ax);
     else
       fprintf (plot_stream, "unset m%stics;\n", ax);
     endif
   endif
 endfunction
 
+function ticklabel = ticklabel_to_cell (ticklabel)
+  if (! isempty (ticklabel) && ! iscell (ticklabel))
+    if (isnumeric (ticklabel))
+      ## Use upto 5 significant digits
+      ticklabel = num2str (ticklabel(:), 5);
+    endif
+    ticklabel = cellstr (ticklabel);
+  endif
+endfunction
+
 function colorspec = get_text_colorspec (color, mono)
   if (mono)
     colorspec = "";
   else
     colorspec = sprintf ("textcolor rgb \"#%02x%02x%02x\"",
 			 round (255*color));
   endif
 endfunction
